From: Prarit Bhargava <prarit@redhat.com>
Date: Fri, 6 Feb 2009 14:14:15 -0500
Subject: [x86] move pci_video_fixup to later in boot
Message-id: 20090206191414.23483.51931.sendpatchset@prarit.bos.redhat.com
O-Subject: [RHEL5 PATCH]: move pci_video_fixup to later in boot
Bugzilla: 467785
RH-Acked-by: Jon Masters <jcm@redhat.com>

During PCI enumeration, the following warning was output in dmesg.

PCI: Ignore bogus resource 6 [0:0] of 0000:01:05.0

Upstream (since at least 2.6.25) has moved pci_fixup_video to a final stage
fixup.  We should do the same to avoid this false bogus (a bogus bogus?)
warning message.

Successfully compiled and tested on a Sun X6240.

Resolves BZ 467785.

diff --git a/arch/i386/pci/fixup.c b/arch/i386/pci/fixup.c
index 83c3645..cb0cbb8 100644
--- a/arch/i386/pci/fixup.c
+++ b/arch/i386/pci/fixup.c
@@ -385,7 +385,7 @@ static void __devinit pci_fixup_video(struct pci_dev *pdev)
 		printk(KERN_DEBUG "Boot video device is %s\n", pci_name(pdev));
 	}
 }
-DECLARE_PCI_FIXUP_HEADER(PCI_ANY_ID, PCI_ANY_ID, pci_fixup_video);
+DECLARE_PCI_FIXUP_FINAL(PCI_ANY_ID, PCI_ANY_ID, pci_fixup_video);
 
 /*
  * Some Toshiba laptops need extra code to enable their TI TSB43AB22/A.
